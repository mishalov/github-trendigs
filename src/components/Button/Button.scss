@import "../../styles/variables.scss";
.button {
  color: #fff;
  background-color: $main-ui-color;
  border-radius: 3px;
  padding: 8px 12px;
  border-style: none;

  &--disabled {
    background-color: $col-secondary-element;
    color: $col-primary-font;
  }
}
