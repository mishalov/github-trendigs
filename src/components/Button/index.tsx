import React from "react";
import "./Button.scss";

interface IButtonProps {
  onClick: () => void;
}

const Button: React.FC<IButtonProps> = props => {
  const { onClick, children } = props;
  return (
    <button className="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
