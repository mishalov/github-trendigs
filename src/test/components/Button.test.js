import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import Button from "../../components/Button";

Enzyme.configure({ adapter: new Adapter() });
const wrappers = [];

beforeEach(() => {
  wrappers.push(shallow(<Button />));
  wrappers.push(shallow(<Button>test</Button>));
  wrappers.push(shallow(<Button status="disabled">test</Button>));
  wrappers.push(shallow(<Button status="success">test</Button>));
  wrappers.push(
    shallow(
      <Button
        text="test button"
        onClick={() => {
          alert("test button pressed");
        }}
      />
    )
  );
});

describe("<Button /> rendering", () => {
  it("renders correctly", () => {
    expect(wrappers[0]).toMatchSnapshot();
  });
  it("with text renders correctly", () => {
    expect(wrappers[1]).toMatchSnapshot();
  });
  it("with text and DISABLED status renders correctly", () => {
    expect(wrappers[2]).toMatchSnapshot();
  });
  it("with text and SUCCESS status renders correctly", () => {
    expect(wrappers[3]).toMatchSnapshot();
  });
  it("primary button with text and onClick renders correctly", () => {
    expect(wrappers[4]).toMatchSnapshot();
  });
});
