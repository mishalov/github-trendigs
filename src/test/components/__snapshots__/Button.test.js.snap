// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering primary button with text and onClick renders correctly 1`] = `
<button
  className="button button--normal"
  onClick={[Function]}
/>
`;

exports[`<Button /> rendering renders correctly 1`] = `
<button
  className="button button--normal"
/>
`;

exports[`<Button /> rendering with text and DISABLED status renders correctly 1`] = `
<button
  className="button button--disabled"
>
  test
</button>
`;

exports[`<Button /> rendering with text and SUCCESS status renders correctly 1`] = `
<button
  className="button button--success"
>
  test
</button>
`;

exports[`<Button /> rendering with text renders correctly 1`] = `
<button
  className="button button--normal"
>
  test
</button>
`;
